Sources/Momento/errors/Errors.swift

```diff
@@ -305,3 +362,19 @@ internal func grpcStatusToSdkError(grpcStatus: GRPCStatus) -> SdkError {
         return UnknownError(message: "Unknown error", innerException: grpcStatus)
     }
 }
+
+func processError<Request: Message & Sendable, Response: Message & Sendable>(
+    err: GRPCStatus,
+    call: UnaryCall<Request, Response>
+) async -> SdkError {
+    if #available(macOS 10.15, *) {
+        do {
+            let trailers = try await call.trailingMetadata.get()
+            return grpcStatusToSdkError(grpcStatus: err, metadata: trailers)
+        } catch {
+            return grpcStatusToSdkError(grpcStatus: err)
+        }
+    } else {
+        return grpcStatusToSdkError(grpcStatus: err)
+    }
+}
```

Comment:
the function signature looks weird, but it allows us to encompass the various unary RPC call types so we can extract trailing metadata in just one place rather than in all the `catch let err as GRPCStatus` branches.

Example call types: `<CacheClient__GetRequest, CacheClient__GetResponse>` and `<CacheClient__SetRequest, CacheClient__SetResponse>`